src/mongoose_user_stats.erl

```diff
@@ -92,6 +117,28 @@ term_to_string(Term) ->
     R= io_lib:format("~p",[Term]),
     lists:flatten(R).
 
-client_id() ->
-    % TODO in the later implementation store client's ID in eg mnesia table and report stats with the same ID for the same client
-    rand:uniform(1000 * 1000 * 1000 * 1000 * 1000).
+maybe_make_and_save_new_client_id() ->
+    T = fun() ->
+        mnesia:first(?TAB_NAME)
+    end,
+    case mnesia:transaction(T) of
+        {atomic, '$end_of_table'} ->
+            make_and_save_new_client_id();
```

Comment:
Still has a race condition window. New ID should be created and stored in the same transaction.